src/daft-sql/src/python.rs

```diff
@@ -45,7 +45,7 @@ pub fn sql_exec(
     session: &PySession,
     config: PyDaftPlanningConfig,
 ) -> PyResult<Option<PyObject>> {
-    if let Some(plan) = execute_statement(session.into(), sql)? {
+    if let Some(plan) = execute_statement(session.inner().clone_ref(), sql)? {
```

Comment:
Did `.into::<Session>().clone_ref()` not work? I suppose `session.session.clone_ref()` would avoid the arc clone in the old From

Reply:
i really dislike `impl From<&T>` in general as it hides the cloning. This is a bit more transparent.

Reply:
There was also a conversation a while back about making the SQLPlanner take in a borrowed session instead of an owned session. I think it'd be best to go that direction anyways. That's definitely out of scope for this PR, but something we should tackle soon as I feel it'd clean up a lot of these unnecessary `into` and `clone`'s.